src/components/BondingCurveChart.tsx

```diff
@@ -0,0 +1,426 @@
+         */
+        if (hasInitGoal && !hasExceededInitGoal) {
+            datasets.push(
+                generateLine(
+                    [points.currentSupply, points.kickstarterEnd],
+                    chartGreen,
+                    'Kickstarter Funded'
+                )
+            );
+
+            datasets.push(
+                generateLine(
+                    [points.kickStarterStart, points.currentSupply],
+                    chartBlue,
+                    'Kickstarter Unfunded'
+                )
+            );
+        } else if (hasInitGoal) {
+            datasets.push(
+                generateLine(
+                    [points.kickStarterStart, points.kickstarterEnd],
+                    chartBlue,
+                    'Kickstarter Funded'
+                )
+            );
+        }
+
+        if (hasExceededInitGoal) {
+            datasets.push(
+                generateLine(
+                    [points.curveStart, points.currentSupply],
+                    chartBlue,
+                    'Curve chart funded'
+                )
+            );
+
+            datasets.push(
+                generateLine(
+                    [points.kickstarterEnd, points.curveStart],
+                    chartBlue,
+                    'Curve chart point of change'
+                )
+            );
+
+            datasets.push(
+                generateLine(
+                    [points.currentSupply, points.maxSupplyToShow],
+                    chartGray,
+                    'Curve chart unfunded'
+                )
+            );
+        } else {
+            datasets.push(
+                generateLine(
+                    [points.kickstarterEnd, points.curveStart],
+                    chartGray,
+                    'Curve chart point of change'
+                )
+            );
+
+            datasets.push(
+                generateLine(
+                    [points.curveStart, points.maxSupplyToShow],
+                    chartGray,
+                    'Curve chart funded'
+                )
+            );
+        }
+
+        datasets.push(generateSupplyMarker(points.currentSupply, 'Current Supply'));
+
+        // TODO: Determine how to display predicted future supply / price after current buy
+        // if (hasActiveInput) {
+        //     datasets.push(generateSupplyMarker(points.futureSupply, 'Future Supply'));
+        // }
+
+        const data = {
+            datasets,
+        };
+
+        return {
+            data,
+            options,
+        };
+    };
+
+    if (requiredDataLoaded) {
+        const generated = generateChart();
+        data = generated.data;
+        options = generated.options;
+    }
+
+    /*
+        Draw a strait line for the inital goal. The PRICE for this is the slope for total supply 0
+        We may have to hardcode this value...
+
+        It's a slope but I DON"T understand because it's 1 / 10^18!!!! And it seems to be about 1/3 of tokens.
```

Comment:
still relevant?

Reply:
Yea... I'll stop writing notes in the code